mypy/checker.py

```diff
@@ -6948,19 +6942,19 @@ def refine_identity_comparison_expression(
         expressions in the chain to a Literal type. Performing this coercion is sometimes
         too aggressive of a narrowing, depending on context.
         """
-        should_coerce = True
-        if coerce_only_in_literal_context:
 
-            def should_coerce_inner(typ: Type) -> bool:
-                typ = get_proper_type(typ)
-                return is_literal_type_like(typ) or (
-                    isinstance(typ, Instance) and typ.type.is_enum
-                )
-
-            should_coerce = any(should_coerce_inner(operand_types[i]) for i in chain_indices)
+        if coerce_only_in_literal_context:
```

Comment:
No logic change to `coerce_only_in_literal_context`, just refactor to avoid unnecessary local function. (I do want to explore logic changes here in the future though)